### PR TITLE
NickAkhmetov/CAT-454 CAT-627 Data Products improvements

### PR DESCRIPTION
**Pull request by NickAkhmetov**
This PR resolves some issues with the data products section:

- The `generated by` logic now excludes portal-containers and formats `pipeline.cwl` workflows' names to use the github repo name instead
- Data products sections with more than 4 entries are now scrollable to prevent them from taking up 80% of the page in outlier cases like SPRM

https://github.com/hubmapconsortium/portal-ui/assets/19957804/41994e12-14ea-4b7b-8269-eecdbd4cca43

I accidentally started this work from `main` then rebased to `dependency-bumps` so the Git history is a bit wonky; the relevant changes are in `DataProducts.tsx` (max height and overflow) and `DataProducts/hooks.ts` ("generated by" revisions)

### CODE REVIEW

File: context/app/static/js/components/detailPage/files/DataProducts/hooks.ts [hunk 1]
```diff
@@ -16,20 +16,42 @@ const defaultPipeline: PipelineInfo = {
...
+}
+
 /**
  * Extract the latest origin and name from the entity data in the flask context
  * @returns {Object} { origin (pipeline URL), name (pipeline name, if specified)}
  */
 function usePipelineInfo(): PipelineInfo {
   const { entity } = useFlaskDataContext();
   const dagList = entity.metadata.dag_provenance_list ?? [];
+  console.log({ dagList });
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/files/DataProducts/hooks.ts:40]:
```suggestion
```

File: context/app/static/js/components/detailPage/files/DataProducts/hooks.ts [hunk 2]
```diff
@@ -16,20 +16,42 @@ const defaultPipeline: PipelineInfo = {
   origin: 'https://github.com/hubmapconsortium/ingest-pipeline.git',
 };
 
+// Values that should not be displayed as pipeline info
+const originBlacklist = ['https://github.com/hubmapconsortium/portal-containers'];
+
+// Undescriptive pipeline names which should be replaced with GitHub repo names
+const nameBlacklist = ['pipeline.cwl'];
+
+function getGithubRepoName(origin: string) {
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/files/DataProducts/hooks.ts:25]: Perhaps add a unit test?

File: context/app/static/js/components/detailPage/files/DataProducts/hooks.ts [hunk 3]
```diff
@@ -16,20 +16,42 @@ const defaultPipeline: PipelineInfo = {
   origin: 'https://github.com/hubmapconsortium/ingest-pipeline.git',
 };
 
+// Values that should not be displayed as pipeline info
+const originBlacklist = ['https://github.com/hubmapconsortium/portal-containers'];
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/files/DataProducts/hooks.ts:20]:
```suggestion
const originDenylist = ['https://github.com/hubmapconsortium/portal-containers'];
```

File: context/app/static/js/components/detailPage/files/DataProducts/hooks.ts [hunk 4]
```diff
@@ -16,20 +16,42 @@ const defaultPipeline: PipelineInfo = {
...
+
+// Undescriptive pipeline names which should be replaced with GitHub repo names
+const nameBlacklist = ['pipeline.cwl'];
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/files/DataProducts/hooks.ts:23]:
```suggestion
const nameDenylist = ['pipeline.cwl'];
```

File: context/app/static/js/components/detailPage/files/DataProducts/hooks.ts [hunk 5]
```diff
@@ -16,20 +16,42 @@ const defaultPipeline: PipelineInfo = {
...
  * @returns {Object} { origin (pipeline URL), name (pipeline name, if specified)}
  */
 function usePipelineInfo(): PipelineInfo {
   const { entity } = useFlaskDataContext();
   const dagList = entity.metadata.dag_provenance_list ?? [];
+  console.log({ dagList });
   // Iterate over the list of DAGs and extract the latest origin and name
   const pipelineInfo = dagList.reduce<PipelineInfo>((acc, dag) => {
     if ('origin' in dag) {
+      if (originBlacklist.includes(dag.origin)) {
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/files/DataProducts/hooks.ts:44]:
```suggestion
      if (originDenylist.includes(dag.origin)) {
```

File: context/app/static/js/components/detailPage/files/DataProducts/hooks.ts [hunk 6]
```diff
@@ -16,20 +16,42 @@ const defaultPipeline: PipelineInfo = {
...
+        return acc;
+      }
       acc.origin = dag.origin;
     }
     if ('name' in dag) {
-      acc.name = dag.name;
+      if (nameBlacklist.includes(dag.name)) {
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/files/DataProducts/hooks.ts:50]:
```suggestion
      if (nameDenylist.includes(dag.name)) {
```

**Review by john-conroy — CHANGES_REQUESTED** [review]
A few changes needed. How did the `console.log` get past the pre-commit hooks?

**Comment by NickAkhmetov** [comment]
> How did the `console.log` get past the pre-commit hooks?

I think my precommit scripts must be messed up, I'll try to get them back into working order - might be time for me to set up a clean environment again...

File: context/app/static/js/components/detailPage/files/DataProducts/hooks.ts (hunk 2 above)
Comment (NickAkhmetov) [reply context/app/static/js/components/detailPage/files/DataProducts/hooks.ts]: I added test cases for this function and pulled out the `.reduce` function as well to make it easier to test in isolation 👍🏻

**Review by john-conroy — APPROVED** [review]
Thanks! The unit tests will be helpful if we have to refer back to that bit of regex in the future.

**Pull request merged**